### ISSUE TITLE
When deploying with CSI - The deployed template isn't at the right VMHW version

### ISSUE THREAD

**Issue by mylesagray** [kubernetes-sigs/cluster-api-provider-vsphere#607]
/kind bug

**What steps did you take and what happened:**
Deploy CAPV cluster with latest manifests, use Ubuntu 18.04 1.15.3 OVA image.

Disk attach operations fail with following error:

![CNS Error](https://dl.dropbox.com/s/kbluoao7tj97peh/Screenshot%202019-10-08%2012.53.59.png)

**What did you expect to happen:**
Disks to attached successfully.

**Anything else you would like to add:**
Unsure if this can be set at an OVA level or if this is required to be done post-deploy of the OVA template?

**Environment:**

- Cluster-api-provider-vsphere version: master
- Kubernetes version: (use `kubectl version`):  1.15.3
- OS (e.g. from `/etc/os-release`): 18.04

**Comment by codyde** [comment]
+1, experiencing the same issue in my vSphere 6.7 u3 environment

**Comment by andrewsykim** [comment]
VMX version update: https://github.com/kubernetes-sigs/image-builder/pull/60

**Comment by codenrhoden** [comment]
I pushed new image last week with the hardware version set to VMX-13. But I only did it for the latest K8s versions -- 1.15.4 and 1.14.7, because I didn't want to overwrite old versions. I don't think we have a policy in place for how to overwrite the old ones (I can do it), but we don't push images with a version other than the K8s version, so it's not clear if an image has been replaced or not.

**Comment by andrewsykim** [comment]
Folks, for now you can update the VMX version post-download of the OVA or use the machine images @codenrhoden mentioned above. Will report back shortly on what's going to happen for the older machine images.

**Comment by andrewsykim** [comment]
So TL;DR on this issue: the new machine images we're pushing will be using VMX-13, but not much we can do about existing images we already released since we don't want to break old versions of CAPV. Some discussions on-going for how to version the machine images here https://github.com/kubernetes-sigs/image-builder/issues/70 so we don't run into this kind of issue again.

@mylesagray has added documentation on how to upgrade the VMX version for our older machine images if you want to use CNS, but in general I think we should just promote the newer images going forward. Does this sound reasonable @mylesagray @codyde?

**Comment by codyde** [comment]
I think thats totally reasonable. I like the documenting of the "how to upgrade" approach for previous images. 

I look at it similar to the manifests image that we talked about previously - a callout to using updated images should help prevent this coming up in the future I suspect. 

All good on my side!

**Comment by andrewsykim** [comment]
Thanks @codyde, closing this for now, but please re-open if you run into other issues or have additional feedback

/close
